src/components/basic_commands.py

```diff
@@ -2828,11 +2830,12 @@ def has_event(transport, idx, to):
         
         if count > 0:
             break;
-        
+
+        to_tmp = 100 - ((transport.last_t - start_t) / 1000);
         to -= 100;
         if to >= 0:
-            transport.wait(100);
-        
+            transport.wait(int(to_tmp));
```

Comment:
Why is this checking for `to >= 0` when the actual waiting time would be in `to_tmp`?

Reply:
Well to>=0 check is re. total time down-count, to_tmp is local to each iteration to signify how much of the 100 was not already spent in call transport.recv(idx, 5, 100).